devtoolsNG/basic/index.html

```diff
@@ -0,0 +1,43 @@
+<body>
+
+  <div id="log"></div> 
+ <input type="button" onclick="myFunction()" value="Clear">
+
+ <script>
+
+   const websocket = new WebSocket('ws://localhost:12345');
+
+   websocket.onopen = e => {
+
+    websocket.send('Hello dear Server!');
+
+     websocket.onmessage = msg => {
+
+      const newEntry = document.createElement('div');
+      newEntry.innerText = msg.data
+      document.getElementById("log").append(newEntry);
+
+     }
+
+   };
+
+   websocket.onerror = e => {
+
+     console.log('No connection found');
+
+   };
+
+   websocket.onclose = e => {
+
+    console.log(`Websocket closing. ${e}`)
+    document.getElementById("log").innerHTML = document.getElementById("log").innerHTML + "<div>WebSocket Closing</div>"
+
+   }
+
+   function myFunction() {
+    document.getElementById("log").innerHTML = ""
+   };

```

Comment:
Seems like there's inconsistent tabbing/spacing here. I'll defer to @piotrswigon on the JavaScript readability though, since it's been years since I've done JS.

Reply:
Yeah, would be good to fix. Also do we really want so many newlines? I think they hurt readability.

Reply:
Updated the formatting and cleaned up the code a bit.